backend/server.js

```diff
@@ -2,27 +2,168 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import crypto from 'crypto'
+import bcrypt from 'bcrypt'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/projectAuth"
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: false, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const User = mongoose.model('User', {
+  username: {
+    type: String,
+    required: true,
+    unique: true
+  },
+  email: {
+    type: String,
+    required: true,
+    unique: true
+  },
+  fullName: {
+    type: String,
+    unique: false
+  },
+  age: {
+    type: Number,
+    unique: false
+  },
+  location: {
+    type: String,
+    unique: false
+  },
+  description: {
+    type: String,
+    unique: false
+  },
+  password: {
+    type: String,
+    required: true
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex')
+  }
```

Comment:
woohoo! great model. In real life, you want to separate login data and personal data to separate models and refer to each other for security reasons.